Remove bashisms from Makefile snippets

diff --git a/Make.coverity b/Make.coverity
index b80b091..456b71b 100644
--- a/Make.coverity
+++ b/Make.coverity
@@ -12,7 +12,7 @@ cov-int : clean

 cov-clean :
 	@rm -vf $(NAME)-coverity-*.tar.*
-	@if [[ -d cov-int ]]; then rm -rf cov-int && echo "removed 'cov-int'"; fi
+	@if [ -d cov-int ]; then rm -rf cov-int && echo "removed 'cov-int'"; fi

 cov-file : | $(COV_FILE)

@@ -20,9 +20,9 @@ $(COV_FILE) : cov-int
 	tar caf $@ cov-int

 cov-upload :
-	@if [[ -n "$(COV_URL)" ]] &&					\
-	    [[ -n "$(COV_TOKEN)" ]] &&					\
-	    [[ -n "$(COV_EMAIL)" ]] ;					\
+	@if [ -n "$(COV_URL)" ] &&					\
+	    [ -n "$(COV_TOKEN)" ] &&					\
+	    [ -n "$(COV_EMAIL)" ] ;					\
 	then								\
 		echo curl --form token=$(COV_TOKEN) --form email="$(COV_EMAIL)" --form file=@"$(COV_FILE)" --form version=$(VERSION).1 --form description="$(COMMIT_ID)" "$(COV_URL)" ; \
 		curl --form token=$(COV_TOKEN) --form email="$(COV_EMAIL)" --form file=@"$(COV_FILE)" --form version=$(VERSION).1 --form description="$(COMMIT_ID)" "$(COV_URL)" ; \
diff --git a/Make.scan-build b/Make.scan-build
index 6cdae21..6a13367 100644
--- a/Make.scan-build
+++ b/Make.scan-build
@@ -11,7 +11,7 @@ SCAN_RANLIB = $(CROSS_COMPILE)llvm-ranlib
 scan-test : ; $(SCAN_BUILD_ERROR)

 scan-clean : clean
-	@if [[ -d scan-results ]]; then rm -rf scan-results && echo "removed 'scan-results'"; fi
+	@if [ -d scan-results ]; then rm -rf scan-results && echo "removed 'scan-results'"; fi

 scan-build : | scan-test
 scan-build : clean

